crates/analytics/src/search.rs

```diff
@@ -22,14 +22,26 @@ pub async fn msearch_results(
         .add_filter_clause("merchant_id".to_string(), merchant_id.to_string())
         .switch()?;
 
-    let response_body = client
+    let response_text = client
         .execute(query_builder)
         .await
         .change_context(OpenSearchError::ConnectionError)?
-        .json::<OpenMsearchOutput>()
+        .text()
         .await
         .change_context(OpenSearchError::ResponseError)?;
 
+    let response_body = match serde_json::from_str::<OpenMsearchOutput>(&response_text) {
+        Ok(parsed_response) => parsed_response,
+        Err(parse_error) => {
+            tracing::error!(
+                "Failed to parse response: {:?}, raw response: {}",
+                parse_error,
+                response_text
+            );
+            return Err(error_stack::Report::from(OpenSearchError::ResponseError));
+        }
```

Comment:
```suggestion
        er@Err(parse_error) => {
            tracing::error!(
                "Failed to parse response: {:?}, raw response: {}",
                parse_error,
                response_text
            );
            return er.change_context(OpenSearchError::ResponseError);
        }
```
can we do something like this ?

Reply:
Can we do it a bit more functional ?

something like

```rs
.text()
.change_context(OpensearchError::ResponseBodyError)
.and_then(|body|
    serde_json::from_str(&body)
    .change_context(OpensearchError::ResponseDeserializationError)
    .attach_printable(&body)
)?
```

Reply:
Done